pkg/scheduler/scheduling_queue_test.go

```diff
@@ -688,11 +694,296 @@ func TestHeadroom(t *testing.T) {
+	//   - parent (max: 20,8)
+	//     - leaf1 (max: 10, 8)  (alloc: 5,3)
+	//     - leaf2 (max: nil)    (alloc: 6,4)
+	resMap = map[string]string{"first": "20", "second": "8"}
+	root, err = createRootQueue(nil)
+	assert.NilError(t, err, "failed to create root queue with limit")
+	parent, err = createManagedQueue(root, "parent", true, resMap)
+	assert.NilError(t, err, "failed to create parent queue")
+	resMap = map[string]string{"first": "10", "second": "8"}
+	leaf1, err = createManagedQueue(parent, "leaf1", false, resMap)
+	assert.NilError(t, err, "failed to create leaf1 queue")
+	leaf2, err = createManagedQueue(parent, "leaf2", false, nil)
+	assert.NilError(t, err, "failed to create leaf2 queue")
+
+	// allocating and allocated
+	res, err = resources.NewResourceFromConf(map[string]string{"first": "1", "second": "1"})
+	assert.NilError(t, err, "failed to create resource")
+	leaf1.incAllocatingResource(res)
+	leaf2.incAllocatingResource(res)
+	leaf2.incAllocatingResource(res)
+	res, err = resources.NewResourceFromConf(map[string]string{"first": "4", "second": "2"})
+	assert.NilError(t, err, "failed to create resource")
+	err = leaf1.QueueInfo.IncAllocatedResource(res, true)
+	assert.NilError(t, err, "failed to set allocated resource on leaf1")
+	err = leaf2.QueueInfo.IncAllocatedResource(res, true)
+	assert.NilError(t, err, "failed to set allocated resource on leaf2")
+
+	// root headroom should be nil
+	headRoom = root.getMaxHeadRoom()
+	assert.Assert(t, headRoom == nil, "headRoom of root should be nil because no max set for all queues")
+
+	// parent headroom = parentMax - leaf1Allocated - leaf2Allocated
+	// parent headroom = (20 - 5 - 6, 8 - 3 - 4) = (9, 1)
+	res, err = resources.NewResourceFromConf(map[string]string{"first": "9", "second": "1"})
+	headRoom = parent.getMaxHeadRoom()
+	if err != nil || !resources.Equals(res, headRoom) {
+		t.Errorf("parent queue head room not as expected %v, got: %v (err %v)", res, headRoom, err)
+	}
+
+	// leaf1 headroom = MIN(parentHeadRoom, leaf1Max - leaf1Allocated)
+	// leaf1 headroom = MIN((9,1), (10-5, 8-3)) = MIN((9,1), (5,5)) = MIN(5, 1)
+	res, err = resources.NewResourceFromConf(map[string]string{"first": "5", "second": "1"})
+	headRoom = leaf1.getMaxHeadRoom()
+	if err != nil || !resources.Equals(res, headRoom) {
+		t.Errorf("leaf1 queue head room not as expected %v, got: %v (err %v)", res, headRoom, err)
+	}
+
+	// leaf2 headroom = parentMax - leaf1Allocated - leaf2Allocated
+	res, err = resources.NewResourceFromConf(map[string]string{"first": "9", "second": "1"})
+	headRoom = leaf2.getMaxHeadRoom()
+	if err != nil || !resources.Equals(res, headRoom) {
+		t.Errorf("leaf2 queue head room not as expected %v, got: %v (err %v)", res, headRoom, err)
+	}
+
+	// recreate the structure, set max capacity in parent and a leaf queue
+	// structure is:
+	// root (max: nil)
+	//   - parent (max: 20,8)
+	//     - parent1 (max: nil)
+	//       - parent2 (max: 10,4)
+	//         - leaf1 (max: nil)   (alloc: 3,3)
```

Comment:
This tests the same as a parent with a leaf below it and does not include any new test case.
please remove this case

Reply:
Removed